src/subscript/co2containment/co2_containment.py

```diff
@@ -0,0 +1,346 @@
+
+
+# pylint: disable-msg=too-many-locals
+def _merge_date_rows(
+    data_frame: pd.DataFrame, calc_type: CalculationType
+) -> pd.DataFrame:
+    """
+    Uses input dataframe to calculate various new columns and renames/merges
+    some columns.
+
+    Args:
+        data_frame (pd.DataFrame): Input data frame
+        calc_type (pd.DataFrame): Input data frame
+        calc_type_input (CalculationType): Choose mass / volume_extent /
+            volume_actual / volume_actual_simple from enum CalculationType
+
+    Returns:
+        pd.DataFrame: Output data frame
+    """
+    print("Merging data rows for data frame")
+    data_frame = data_frame.drop("zone", axis=1)
+    # Total
+    df1 = (
+        data_frame.drop(["phase", "location"], axis=1)
+        .groupby(["date"])
+        .sum()
+        .rename(columns={"amount": "total"})
+    )
+    total_df = df1.copy()
+    if calc_type == CalculationType.VOLUME_EXTENT:
+        df2 = data_frame.drop("phase", axis=1).groupby(["location", "date"]).sum()
+        df2a = df2.loc[("contained",)].rename(columns={"amount": "total_contained"})
+        df2b = df2.loc[("outside",)].rename(columns={"amount": "total_outside"})
+        df2c = df2.loc[("hazardous",)].rename(columns={"amount": "total_hazardous"})
+        for _df in [df2a, df2b, df2c]:
+            total_df = total_df.merge(_df, on="date", how="left")
+    else:
+        df2 = data_frame.drop("location", axis=1).groupby(["phase", "date"]).sum()
+        df2a = df2.loc["gas"].rename(columns={"amount": "total_gas"})
+        df2b = df2.loc["aqueous"].rename(columns={"amount": "total_aqueous"})
+        # Total by containment
+        df3 = data_frame.drop("phase", axis=1).groupby(["location", "date"]).sum()
+        df3a = df3.loc[("contained",)].rename(columns={"amount": "total_contained"})
+        df3b = df3.loc[("outside",)].rename(columns={"amount": "total_outside"})
+        df3c = df3.loc[("hazardous",)].rename(columns={"amount": "total_hazardous"})
+        # Total by containment and phase
+        df4 = data_frame.groupby(["phase", "location", "date"]).sum()
+        df4a = df4.loc["gas", "contained"].rename(columns={"amount": "gas_contained"})
+        df4b = df4.loc["aqueous", "contained"].rename(
+            columns={"amount": "aqueous_contained"}
+        )
+        df4c = df4.loc["gas", "outside"].rename(columns={"amount": "gas_outside"})
+        df4d = df4.loc["aqueous", "outside"].rename(
+            columns={"amount": "aqueous_outside"}
+        )
+        df4e = df4.loc["gas", "hazardous"].rename(columns={"amount": "gas_hazardous"})
+        df4f = df4.loc["aqueous", "hazardous"].rename(
+            columns={"amount": "aqueous_hazardous"}
+        )
+        for _df in [df2a, df2b, df3a, df3b, df3c, df4a, df4b, df4c, df4d, df4e, df4f]:
+            total_df = total_df.merge(_df, on="date", how="left")
+    return total_df.reset_index()
+
+
+def make_parser() -> argparse.ArgumentParser:
```

Comment:
Not super important, but consider renaming to `get_parser()` as this is what the vast majority of other subscripts call it. Doesn't matter so much so long as the args receiver in the documentation (if used) has the right name.